### PR TITLE
rsdk-1976 add constraint messages to api

### PR DESCRIPTION
**Pull request by biotinker**

### CODE REVIEW

File: proto/viam/service/motion/v1/motion.proto [hunk 1]
```diff
@@ -74,3 +75,33 @@ message GetPoseRequest {
...
+}
+
+// OrientationConstraint specifies that the component being moved will not deviate its orientation beyond some threshold relative to the goal. It does not constrain the motion of components other than the `component_name` specified in motion.Move
+message OrientationConstraint {
+  optional float orientation_tolerance_degs = 1; // Max allowable orientation deviation, in degrees, while on the shortest path between start / goal states
+}
+
+// CollisionSpecification is used to selectively apply obstacle avoidance to specific parts of the robot
+message CollisionSpecification {
+  message AllowedFrameCollisions {
```
Comment (raybjork) [inline proto/viam/service/motion/v1/motion.proto:101]: [q] I know this was part of the scope, but I'm curious about the decision to have this as a nested message vs a top level message?

**Review by raybjork — APPROVED** [review]
Just a question, but LGTM

**Review by spieswl — APPROVED** [review]
LGTM

File: proto/viam/service/motion/v1/motion.proto (hunk 1 above)
Comment (biotinker) [reply proto/viam/service/motion/v1/motion.proto:101]: Admittedly this creates long, ugly autogenerated code.

However, it prevents this message being used by many different things that want pairings of things and its eventual expansion.

**Pull request merged**